subsys/bluetooth/host/adv.c

```diff
@@ -1171,6 +1191,22 @@ static int le_ext_adv_param_set(struct bt_le_ext_adv *adv,
 	if (param->options & BT_LE_ADV_OPT_CODED) {
 		cp->prim_adv_phy = BT_HCI_LE_PHY_CODED;
 		cp->sec_adv_phy = BT_HCI_LE_PHY_CODED;
+
+#if defined(CONFIG_BT_EXT_ADV_CODING_SELECTION)
+		if (opcode == BT_HCI_OP_LE_SET_EXT_ADV_PARAM_V2) {
+			/* Default to no required coding scheme */
+			uint8_t adv_phy_opt = BT_HCI_LE_ADV_PHY_OPTION_NO_REQUIRED;
+
+			if (param->options & BT_LE_ADV_OPT_REQUIRE_S8_CODING) {
+				adv_phy_opt = BT_HCI_LE_ADV_PHY_OPTION_REQUIRE_S8;
+			} else if (param->options & BT_LE_ADV_OPT_REQUIRE_S2_CODING) {
+				adv_phy_opt = BT_HCI_LE_ADV_PHY_OPTION_REQUIRE_S2;
+			}
```

Comment:
```suggestion
			uint8_t adv_phy_opt;

			if (param->options & BT_LE_ADV_OPT_REQUIRE_S8_CODING) {
				adv_phy_opt = BT_HCI_LE_ADV_PHY_OPTION_REQUIRE_S8;
			} else if (param->options & BT_LE_ADV_OPT_REQUIRE_S2_CODING) {
				adv_phy_opt = BT_HCI_LE_ADV_PHY_OPTION_REQUIRE_S2;
			} else {
				adv_phy_opt = BT_HCI_LE_ADV_PHY_OPTION_NO_REQUIRED;
			}
```

Reply:
`_PREFERS_` can be a not supported, but do not see passing them over HCI is not too much of work in this PR.